File: client/src/app/entities/vehicle.ts
import {User} from './user';
import {DataPage} from '../shared/data-page';
import {Vehicletype} from './vehicletype';
import {Customer} from './customer';
import {Vehiclebrand} from './vehiclebrand';
import {Transmission} from './transmission';
import {Fueltype} from './fueltype';

export class Vehicle {
  id: number;
  code: string;
  tocreation: string;
  description: string;
  creator: User;
  no: string;
  photo: string;
  vehicletype: Vehicletype;
  makeyear: number;
  customer: Customer;
  vehiclebrand: Vehiclebrand;
  model: string;
  transmission: Transmission;
  fueltype: Fueltype;

  constructor(id: number = null) {
    this.id = id;
  }
}

export class VehicleDataPage extends DataPage{
    content: Vehicle[];
}

File: server/src/main/java/bit/project/server/entity/Vehicle.java
package bit.project.server.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Vehicle {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Integer id;
    private LocalDateTime tocreation;
    private String description;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;

    private String no;
    private String photo;

    @ManyToOne
    private Vehicletype vehicletype;

    private Integer makeyear;

    @ManyToOne
    private Customer customer;

    @ManyToOne
    private Vehiclebrand vehiclebrand;

    private String model;

    @ManyToOne
    private Transmission transmission;

    @ManyToOne
    private Fueltype fueltype;

    public Vehicle(Integer id) {
        this.id = id;
    }

    public Vehicle(Integer id, String no) {
        this.id = id;
        this.no = no;
    }
}
